source/m2minterfaceimpl.cpp

```diff
@@ -404,25 +415,33 @@ void M2MInterfaceImpl::socket_error(uint8_t error_code, bool retry)
     }
 #endif
 
+    char error_code_des[24];
```

Comment:
and seems that there is duplicate strings so maybe those should be defined somewhere else to save a bit more memory